nicegui/elements/qtable.py

```diff
@@ -0,0 +1,67 @@
+from typing import Optional
+
+from typing_extensions import Literal
+
+from ..dependencies import register_component
+from ..element import Element
+
+register_component('q_table', __file__, 'qtable.js')
+
+
+class QTable(Element):
+    def __init__(
+        self,
+        columns: Optional[dict] = [],
+        rows: Optional[list] = [],
+        title: Optional[str] = None,
+        selection_mode: Optional[Literal['single', 'multiple', 'none']] = 'none',
+        selection_key: Optional[str] = '',
+        filter_config: Optional[dict] = {
```

Comment:
May I suggest to not use mutable data structure as default argument ? 
_see flake8-bugfear rule "B006 Do not use mutable data structures for argument defaults "
here for details: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/_

Plus, see the 'v-if' comment : allow None by default would allow to remove the slot which is not possible here.